### PR TITLE
Header include minimization

### PR DESCRIPTION
**Pull request by mcmlevi**
All header files in the main OpenGD project now only include the files they actually need and forward declare the rest. This also then includes the needed files in the .cpp files wherever needed. This should help improve compile times as well as reduce how often specific files need to be recompiled when changes get made as the dependencies are as minimal as possible.

### CODE REVIEW

File: Source/PlayLayer.cpp [hunk 1]
```diff
@@ -40,7 +45,7 @@ ax::Node* cameraFollow;
 
 Scene* PlayLayer::scene(GJGameLevel* level)
 {
-	return LevelDebugLayer::scene(level);
+	//return LevelDebugLayer::scene(level);
```
Comment (mcmlevi) [inline Source/PlayLayer.cpp:48]: This might not be something we actually want submitted

**Comment by mcmlevi** [comment]
https://github.com/Open-GD/OpenGD/pull/44/commits/e3e6b0d9cc99c6f2c99ce61646ca772d994a707b#r1181125564
This might not be something we want to submit, but ill leave that up to the review, Id be happy enough to uncomment that line

**Comment by SergeyMC9730** [comment]
This pull request breaks GCC and Clang support. Please, test your code before sending PL.

**Comment by mcmlevi** [comment]
Yeah my bad, it was a rather straightforward refractor and I was not expecting it to break on GCC and clang. 
https://stackoverflow.com/questions/71416/forward-declaring-an-enum-in-c/1280969#1280969

I know forward-declaring enums was possible however I missed the fact that forward-declaring a regular enum is actually not something that is possible without specifying the size. The fix should be rather straight forward and can be something I can make in a bit what will be changing the enum to an enum class and the forward declarations as well.

**Pull request merged**